### PR TITLE
feat: Add breadcrumb component

### PR DESCRIPTION
**Pull request by SantiagoRodH**
Screenshot 2024-05-31 at 12 42 07 PM

- In Home page
Screenshot 2024-05-31 at 12 39 42 PM

- Hide in navigation option in Overview page  set to true
Screenshot 2024-05-31 at 12 39 49 PM

### CODE REVIEW

File: ui.apps/src/main/content/jcr_root/apps/aem-bootcamp/components/structure/breadcrumb/.content.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:cq="http://www.day.com/jcr/cq/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    jcr:primaryType="cq:Component"
+    jcr:title="Breadcrumb Component"
```
Comment (lflondonol92) [inline ui.apps/src/main/content/jcr_root/apps/aem-bootcamp/components/structure/breadcrumb/.content.xml:4]: the jcr:description is missing

File: core/src/main/java/com/aembootcamp/core/models/BreadcrumbModel.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  protected static final String RESOURCE_TYPE = "aem-bootcamp/components/structure/breadcrumb";
+
+  @ScriptVariable
+  private Page currentPage;
+
+  @Delegate @Self @Via(type = ResourceSuperType.class)
+  private Breadcrumb breadcrumb;
+
+  public Boolean getIsHome() {
+    if(currentPage != null) {
```
Comment (lflondonol92) [inline core/src/main/java/com/aembootcamp/core/models/BreadcrumbModel.java:26]: Use the template of the currentPage and not the path to verify this.

File: core/src/main/java/com/aembootcamp/core/models/BreadcrumbModel.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  protected static final String RESOURCE_TYPE = "aem-bootcamp/components/structure/breadcrumb";
+
+  @ScriptVariable
+  private Page currentPage;
+
+  @Delegate @Self @Via(type = ResourceSuperType.class)
+  private Breadcrumb breadcrumb;
+
+  public Boolean getIsHome() {
```
Comment (lflondonol92) [inline core/src/main/java/com/aembootcamp/core/models/BreadcrumbModel.java]: rename this method. getIs is a bad naming strategy. Please rename it to isHomePage.

File: ui.apps/src/main/content/jcr_root/apps/aem-bootcamp/components/structure/breadcrumb/breadcrumb.html [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<sly data-sly-use.breadcrumb="com.aembootcamp.core.models.BreadcrumbModel" />
+<sly data-sly-test="${!breadcrumb.isHome}" data-sly-use.template="core/wcm/components/commons/v1/templates.html">
```
Comment (lflondonol92) [inline ui.apps/src/main/content/jcr_root/apps/aem-bootcamp/components/structure/breadcrumb/breadcrumb.html]: breadcrumb.homePage

**Review by lflondonol92 — CHANGES_REQUESTED** [review]

**Review by joseVperficient — APPROVED** [review]

**Pull request merged**